packages/tracing/src/transaction.ts

```diff
@@ -117,6 +112,11 @@ export class Transaction extends SpanClass implements TransactionInterface {
       }).endTimestamp;
     }
 
+    // ensure that we have a tracestate to attach to the envelope header
+    if (!this.metadata.tracestate?.sentry) {
+      this.metadata.tracestate = { ...this.metadata.tracestate, sentry: this._getNewTracestate() };
+    }
+
```

Comment:
Because it needs to get through to [here](https://github.com/getsentry/sentry-javascript/blob/2a7b830938aeaef8acc66a3b5b4a902e34e79df6/packages/core/src/request.ts#L67) somehow, and putting it in `debug_meta` means that even if it slips through* and makes it to sentry, it won't show up for the user. (Previously the information hitched a ride on a custom tag, and a bug in the code meant that it was showing up for users.) 

\*If you look a few lines into the linked function, you'll see that in fact it gets pulled out of the eventual event payload. Same with the sampling data.

As for it being a private method, is there any advantage to that over it being protected (which it is currently)?

Reply:
Again it was just a nit, but in general I find `protected` members to be sign of a messy abstraction. It is something that you think you shouldn't expose ( otherwise you would make it public) but have to expose it "a little" because derived classes need to mess with it.